core/src/main/scala/kafka/server/KafkaRaftServer.scala

```diff
@@ -181,7 +181,7 @@ object KafkaRaftServer {
     }
 
     // Load the BootstrapMetadata.
-    val bootstrapDirectory = new BootstrapDirectory(config.metadataLogDir)
+    val bootstrapDirectory: LegacyBootstrapDirectory = new LegacyBootstrapDirectory(config.metadataLogDir)
```

Comment:
```suggestion
    val bootstrapDirectory = new LegacyBootstrapDirectory(config.metadataLogDir)
```
That should work right?